python/hail/tests/test_utils.py

```diff
@@ -54,6 +54,40 @@ def test_hadoop_methods(self):
         with self.assertRaises(Exception):
             hadoop_open('/tmp/randomBytesOut', 'xb')
 
+    def test_hadoop_exists(self):
+        self.assertTrue(hl.hadoop_exists(resource('ls_test')))
+        self.assertFalse(hl.hadoop_exists(resource('doesnt.exist')))
+
+    def test_hadoop_is_file(self):
+        self.assertTrue(hl.hadoop_is_file(resource('ls_test/f_50')))
+        self.assertFalse(hl.hadoop_is_file(resource('ls_test/subdir')))
+
+    def test_hadoop_stat(self):
+        path1 = resource('ls_test')
+        stat1 = hl.hadoop_stat(path1)
+        self.assertEqual(stat1['is_dir'], True)
```

Comment:
Even if we can't test the values of all the fields, we could assert that the key set is what we expect. Also in hadoop_ls.

Reply:
good call.